src/macaron/parsers/actionparser.py

```diff
@@ -56,15 +63,83 @@ def parse(workflow_path: str, macaron_path: str = "") -> dict:
         subprocess.TimeoutExpired,
         FileNotFoundError,
     ) as error:
-        logger.error("Error while parsing GitHub Action workflow %s: %s", workflow_path, error)
-        return {}
+        raise ParseError(f"Error while parsing GitHub Action workflow {workflow_path}") from error
 
     try:
         if result.returncode == 0:
             parsed_obj: dict = json.loads(result.stdout.decode("utf-8"))
             return parsed_obj
-        logger.error("GitHub Actions parser failed: %s", result.stderr)
-        return {}
+        raise ParseError(f"GitHub Actions parser failed: {result.stderr.decode('utf-8')}")
     except json.JSONDecodeError as error:
-        logger.error("Error while loading the parsed Actions workflow: %s", error)
-        return {}
+        raise ParseError("Error while loading the parsed Actions workflow") from error
+
+
+def validate_step(step: dict[str, Any]) -> dict | None:
+    """Validate the parsed GitHub Action step.
+
+    Parameters
+    ----------
+    step: dict[str, Any]
+        The parsed step object.
+
+    Returns
+    -------
+    dict | None
+        The Exec object in the parsed step object..
+    """
+    # This validation function can be extended over time based on our use cases.
+    if "Exec" not in step:
+        return None
+
+    return cast(dict, step["Exec"])
+
+
+def validate_run_step(step: dict[str, Any]) -> str | None:
```

Comment:
Is this function intended to validate if the schema of the parsed step obj is as expected and return the value at ` step["Exec"]["Run"]["Value"]`? 
If that is true, I think the name of the function is a bit ambiguous as it only mentions the first action that this function does. 
(Please let me know if I am missing anything)
 I have two suggestions:
1. separate the validation from the extract and return of value. This way, this function will perform exactly how its name indicates
```python
def validate_run_step(step: dict[str, Any]) -> bool:
    if validate_step(step=step) is None:
        return False
    if step["Exec"].get("Run") is None:
        return False
    if step["Exec"]["Run"].get("Value") is None:
        return False

    return True
```
And the handling of data at `step["Exec"]["Run"]["Value"]` can happen outside of this function, by the caller separately. 
```python
if not validate_run_step(parsed_obj):
    raise CallGraphError(f"Invalid run step at {source_path}.")
run_script = str(parsed_obj["Exec"]["Run"]["Value"])
```
This example usage was modified based on
https://github.com/oracle/macaron/blob/ed96daea0f0c4ae81f6f42b464d859f0829563aa/src/macaron/parsers/bashparser.py#L209-L212
2. Rename this function and its doc string to explicitly indicate its two action:
```python
def extract_run_step(step: dict[str, Any]) -> str | None:
...
```
In additions, I think it would also be good to specify what `None` as a return result means (e.g. `None` when `step` object doesn't contain the value of the inlined run script.

Reply:
See commit [cbe9501](https://github.com/oracle/macaron/pull/673/commits/cbe95015233e4971e9f8a9967f61b37ad9ebbe2b)